streams/integration-tests/src/test/java/org/apache/kafka/streams/integration/KTableKTableForeignKeyJoinIntegrationTest.java

```diff
@@ -232,57 +236,134 @@ public void doJoinFromLeftThenDeleteLeftEntity(final boolean leftJoin,
+                    outputTopic.readKeyValuesToList(),
+                    is(List.of(
+                        new KeyValue<>("lhs3", "(lhsValue3|rhs1,rhsValue1)")
                     ))
                 );
                 if (rejoin) {
                     assertThat(
-                        rejoinOutputTopic.readKeyValuesToMap(),
-                        is(mkMap(
-                            mkEntry("lhs3", "rejoin((lhsValue3|rhs1,rhsValue1),lhsValue3|rhs1)")
+                        rejoinOutputTopic.readKeyValuesToList(),
+                        is(List.of(
+                            new KeyValue<>("lhs3", "rejoin((lhsValue3|rhs1,rhsValue1),lhsValue3|rhs1)")
                         ))
                     );
                 }
                 if (materialized) {
                     assertThat(
-                        asMap(store),
-                        is(mkMap(
-                            mkEntry("lhs1", "(lhsValue1|rhs1,rhsValue1)"),
-                            mkEntry("lhs2", "(lhsValue2|rhs2,rhsValue2)"),
-                            mkEntry("lhs3", "(lhsValue3|rhs1,rhsValue1)")
+                        makeList(store),
+                        is(List.of(
+                            new KeyValue<>("lhs1", "(lhsValue1|rhs1,rhsValue1)"),
+                            new KeyValue<>("lhs2", "(lhsValue2|rhs2,rhsValue2)"),
+                            new KeyValue<>("lhs3", "(lhsValue3|rhs1,rhsValue1)")
                         ))
                     );
                 }
             }
+
             // Now delete one LHS entity such that one delete is propagated down to the output.
 
             left.pipeInput("lhs1", null, baseTimestamp + 6);
             assertThat(
-                outputTopic.readKeyValuesToMap(),
-                is(mkMap(
-                    mkEntry("lhs1", null)
+                outputTopic.readKeyValuesToList(),
+                is(List.of(
+                    new KeyValue<>("lhs1", null)
                 ))
             );
             if (rejoin) {
                 assertThat(
-                    rejoinOutputTopic.readKeyValuesToMap(),
-                    is(mkMap(
-                        mkEntry("lhs1", null)
+                    rejoinOutputTopic.readKeyValuesToList(),
+                    hasItem(
+                        KeyValue.pair("lhs1", null))
+                );
+            }
+            if (materialized) {
+                assertThat(
+                    makeList(store),
+                    is(List.of(
+                        new KeyValue<>("lhs2", "(lhsValue2|rhs2,rhsValue2)"),
+                        new KeyValue<>("lhs3", "(lhsValue3|rhs1,rhsValue1)")
                     ))
                 );
             }
+        }
+
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("testCases")
+    public void doJoinFromLeftThenUpdateFkThenRevertBack(final boolean leftJoin,
+        final String optimization,
```

Comment:
nit formatting: should have same indent as first parameter:
```
    public void doJoinFromLeftThenUpdateFkThenRevertBack(final boolean leftJoin,
                                                         final String optimization,
```

Same for other parameters below.